ee/clickhouse/queries/clickhouse_retention.py

```diff
@@ -193,151 +99,120 @@ def process_breakdown_table_result(
-        )
-        default_event_query = (
-            DEFAULT_REFERENCE_EVENT_UNIQUE_PEOPLE_PER_PERIOD_SQL
-            if is_first_time_retention
-            else DEFAULT_REFERENCE_EVENT_PEOPLE_PER_PERIOD_SQL
-        ).format(
-            target_query=target_query_formatted,
-            filters=prop_filters,
-            trunc_func=trunc_func,
-            GET_TEAM_PERSON_DISTINCT_IDS=GET_TEAM_PERSON_DISTINCT_IDS,
+    def _retrieve_people_in_period(self, filter: RetentionPeopleRequest, team: Team):
+        people_appearances = get_people_appearances(filter=filter, team=team)
+        people = get_persons_by_uuids(
+            team_id=team.pk, uuids=[people_appearance.person_id for people_appearance in people_appearances]
         )
 
-        date_from = filter.date_from + filter.selected_interval * filter.period_increment
-        date_to = filter.date_to
-
-        filter = filter.with_data({"total_intervals": filter.total_intervals - filter.selected_interval})
-
-        # NOTE: I'm using `Any` here to avoid typing issues when trying to iterate.
-        query_result: Any = sync_execute(
-            RETENTION_PEOPLE_PER_PERIOD_SQL.format(
-                returning_query=return_query_formatted,
-                filters=prop_filters,
-                first_event_sql=first_event_sql,
-                first_event_default_sql=default_event_query,
-                trunc_func=trunc_func,
-                GET_TEAM_PERSON_DISTINCT_IDS=GET_TEAM_PERSON_DISTINCT_IDS,
-            ),
-            {
-                "team_id": team.pk,
-                "start_date": date_from.strftime(
-                    "%Y-%m-%d{}".format(" %H:%M:%S" if filter.period == "Hour" else " 00:00:00")
-                ),
-                "end_date": date_to.strftime(
-                    "%Y-%m-%d{}".format(" %H:%M:%S" if filter.period == "Hour" else " 00:00:00")
-                ),
-                "offset": filter.offset,
-                "limit": 100,
-                "period": period,
-                **target_params,
-                **return_params,
-                **prop_filter_params,
-            },
-        )
-
-        people_appearances = [
-            AppearanceRow(person_id=row[0], appearance_count=row[1], appearances=row[2]) for row in query_result
-        ]
+        people = people.prefetch_related(Prefetch("persondistinctid_set", to_attr="distinct_ids_cache"))
 
         from posthog.api.person import PersonSerializer
 
-        people = get_persons_by_uuids(team_id=team.pk, uuids=[val[0] for val in query_result])
-        people = people.prefetch_related(Prefetch("persondistinctid_set", to_attr="distinct_ids_cache"))
-
         people_dict = {str(person.uuid): PersonSerializer(person).data for person in people}
 
         result = self.process_people_in_period(
             filter=filter, people_appearances=people_appearances, people_dict=people_dict
         )
         return result
+
+
+def build_actor_query(
```

Comment:
another naming issue. once you merge and resolve conflicts with the clickhous retention actor class, it will be confusing to differentiate this actor query from the other